lib/prefab/criteria_evaluator.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module Prefab
+  class CriteriaEvaluator
+    LOOKUP_KEY = 'LOOKUP'
+    NAMESPACE_KEY = 'NAMESPACE'
+    NO_MATCHING_ROWS = [].freeze
+
+    def initialize(config, project_env_id:, resolver:, base_client:)
+      @config = config
+      @project_env_id = project_env_id
+      @resolver = resolver
+      @base_client = base_client
+    end
+
+    def evaluate(properties)
+      # TODO: optimize this and perhaps do it elsewhere
+      props = properties.transform_keys(&:to_s)
+
+      matching_environment_row_values.each do |conditional_value|
+        # TODO: merge in properties from the environment_row?
```

Comment:
@jdwyah should we handle this now or later?

Reply:
Later ok imo.